Google.PowerShell/BigQuery/GcbqDataset.cs

```diff
@@ -440,10 +460,22 @@ public class RemoveGcbqDataset : GcbqCmdlet
         protected override void ProcessRecord()
         {
             // Form and send request.
-            DatasetsResource.DeleteRequest request = Service.Datasets.Delete(Project, 
-                InputObject.DatasetReference.DatasetId);
+            DatasetsResource.DeleteRequest request;
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.ByObject:
+                    Project = InputObject.ProjectId;
+                    Dataset = InputObject.DatasetId;
+                    request = Service.Datasets.Delete(Project, Dataset);
+                    break;
+                case ParameterSetNames.ByValues:
+                    request = Service.Datasets.Delete(Project, Dataset);
+                    break;
+                default:
+                    throw UnknownParameterSetException;
+            }
             String response = "Dataset Removal Stopped.";
-            if (ShouldProcess(InputObject.DatasetReference.DatasetId))
+            if (ShouldProcess(Dataset))
```

Comment:
Can you add a test to check that if you run the cmdlet with -Whatif then the data is not deleted?